PayPal/src/main/java/com/braintreepayments/api/PayPalLauncher.java

```diff
@@ -0,0 +1,67 @@
+package com.braintreepayments.api;
+
+import static com.braintreepayments.api.BraintreeRequestCodes.PAYPAL;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.FragmentActivity;
+
+/**
+ * Responsible for launching PayPal user authentication in a web browser
+ */
+public class PayPalLauncher {
+
+    private final BrowserSwitchClient browserSwitchClient;
+
+    /**
+     * Used to launch the PayPal flow in a web browser and deliver results to your Activity
+     */
+    public PayPalLauncher() {
+        this(new BrowserSwitchClient());
+    }
+
+    @VisibleForTesting
+    PayPalLauncher(@NonNull BrowserSwitchClient browserSwitchClient) {
+        this.browserSwitchClient = browserSwitchClient;
+    }
+
+    /**
+     * Launches the PayPal flow by switching to a web browser for user authentication
+     *
+     * @param activity       an Android {@link FragmentActivity}
+     * @param payPalResponse the result of the PayPal web authentication flow received from invoking
+     *                       {@link PayPalClient#tokenizePayPalAccount(FragmentActivity,
+     *                       PayPalRequest, PayPalFlowStartedCallback)}
+     * @throws BrowserSwitchException if there is an error launching the PayPal web flow
+     */
+    public void launch(@NonNull FragmentActivity activity, @NonNull PayPalResponse payPalResponse)
+            throws BrowserSwitchException {
```

Comment:
Should this throw or how should this exception be handled? The Activity result based `launch` methods (Google Pay, Venmo) don't throw or callback any exception, so I opted to throw to keep the `launch` parameters cleaner. This does mean a try/catch block in the merchant code (see migration guide example). Other alternative would be to callback an error in this method, or give the `PayPalLauncher` a callback in the constructor to handle results (this would align with the Activity based launchers)

Reply:
Yeah we should design it to have the same API as the Activity launchers if possible. It'd be awesome if we're able to offer a familiar DX for onboarding each Activity and Browser-based payment flow.

Reply:
Ok updated this so that the launcher constructor takes a callback as a parameter (to align with Activity Result flows) and if there's an error in the `launch` method, that error will be delivered to the callback from the constructor.